lib/internal/Magento/Framework/Mail/Template/TransportBuilder.php

```diff
@@ -442,4 +460,28 @@ private function addAddressByType(string $addressType, $email, ?string $name = n
             $this->messageData[$addressType] = $convertedAddressArray;
         }
     }
+
+    /**
+     * Adds attachment part to email
+     *
+     * @param string|null $content
+     * @param string|null $fileName
+     * @param string|null $fileType
+     *
+     * @return $this
+     */
+    public function addAttachment(?string $content, ?string $fileName, ?string $fileType)
```

Comment:
Hello @dudzio12 ,
thank you for your contribution!

According to: https://github.com/laminas/laminas-mime/blob/2.9.x/src/Part.php @ public function setContent($content)

if (! is_string($content) && ! is_resource($content)) it will throw an Exception\InvalidArgumentException, you should avoid ?string $content in parameter and add Exception\InvalidArgumentException in docs

also

according to https://github.com/laminas/laminas-mime/blob/2.9.x/src/Part.php @ public function setType($type = Mime::TYPE_OCTETSTREAM) could be a good idea set $fileType as default Mime::TYPE_OCTETSTREAM to avoid breaking changes after an update of laminas

Reply:
Thanks for diving into the topic, I willl work on that.

Reply:
Added `@throws InvalidArgumentException` annotation and changed argument type from `?string` to `string` for first two arguments.  I have also set `Mime::TYPE_OCTETSTREAM` as default value `$fileType` argument.